htmresearch/frameworks/nlp/classify_document_fingerprint.py

```diff
@@ -251,18 +258,43 @@ def classifyText(self, token, reset=0):
                             sequenceId=-1, reset=0)
       self.network.run(1)
 
-      if reset == 1:
-        self.reset()
-
       if self.verbosity >= 2:
         print "Classifying document:",document
         self.printRegionOutputs()
 
       self.currentDocument = None
-      return self.classifierRegion.getOutputData("categoriesOut")[0:self.numLabels]
+      categoryVotes = self.classifierRegion.getOutputData(
+          "categoriesOut")[0:self.numLabels]
```

Comment:
Here you use `numLabels` to cap the number of inferred categories, but the base class `inferDocument()` uses it specify the total number of possible categories in the dataset. It's not clear why this is needed here.
And could just be `...[:self.numLabels]`